notebook/services/kernels/handlers.py

```diff
@@ -29,7 +29,11 @@ class MainKernelHandler(APIHandler):
     @gen.coroutine
     def get(self):
         km = self.kernel_manager
-        kernels = yield gen.maybe_future(km.list_kernels())
+        res = km.list_kernels()
+        if not isinstance(res, list):
+            kernels = res
+        else:
+            kernels = yield res
```

Comment:
Should these statements be reversed?  Seems like a `list` is the end goal here and anything but that will likely be a future to yield.  I apologize if my understanding is not correct and, if not, perhaps a comment would help.